nifi/nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/main/java/org/apache/nifi/processors/kite/JSONFileReader.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2015 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.kite;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import javax.annotation.Nullable;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.kitesdk.data.DatasetIOException;
+import org.kitesdk.data.spi.AbstractDatasetReader;
+import org.kitesdk.data.spi.DataModelUtil;
+import org.kitesdk.data.spi.JsonUtil;
+import org.kitesdk.data.spi.ReaderWriterState;
+
+/**
+ * This is a temporary addition. The version in 0.18.0 throws a NPE when the
+ * InputStream constructor is used.
+ */
+public class JSONFileReader<E> extends AbstractDatasetReader<E> {
```

Comment:
What happens if someone is using this version of the reader but drops in a later version of Kite?

Can we make this a non-public class that isn't in the package space of Kite? Could we make it a smaller class that inherited from the Kite class and corrected just the specific behavior we're trying to avoid?